pkg/util/billing/billing.go

```diff
@@ -164,6 +164,11 @@ func (m *manager) createOrUpdateE2EBlob(ctx context.Context, doc *api.BillingDoc
 		containerName = containerName[:63]
 	}
 
+	// The following is added to get rid of the '-' at the end in order to avoid an invalid container name.
+	if containerName[len(containerName)-1:] == "-" {
+		containerName = containerName[:len(containerName)-1]
+	}
```

Comment:
I think it would be more idiomatic to do it like so:

```suggestion
	containerName = strings.TrimSuffix(containerName, "-")
```

Reply:
agreed!